### PR TITLE
Update PR template

### LINKED ISSUE #84: Add instructions for writing release-notes in PR.
/kind documentation

Add instructions for writing release-notes in PR. 

We are using [Kubernetes Release Notes Generator](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md#kubernetes-release-notes-generator) to generate our release notes. We should incorporate the instructions for how to write them in the PR template.

/help-wanted
/good-first-issue

### PR DESCRIPTION
**Pull request by prashantrewar**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR is needed to clarify the importance of picking the right kind and writing clear release notes.

**Which issue(s) this PR fixes**:

Fixes #84 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+<!-- If this is your first PR, welcome! Please make sure you read the [contributing guidelines](https://github.com/kubernetes-sigs/ingress2gateway/blob/main/CONTRIBUTING.md). -->
+
+<!-- The release notes and the kind that will be used to generate the right Changelog for the release. To make sure your contribution is recognized, please label this pull request according to what type of issue you are addressing and add the release notes when necessary (see ../CONTRIBUTING.md) -->
```
Comment (LiorLieberman) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
<!-- The release notes and the kind will be used to generate the changelog for the release. To make sure your contribution is recognized, please label this pull request according to what type of issue you are addressing and add the release notes when necessary (see ../CONTRIBUTING.md) -->
```

**Review by LiorLieberman — COMMENTED** [review]
awesome! I appreciate your help with this

/approve

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (prashantrewar) [reply .github/PULL_REQUEST_TEMPLATE.md]: done!

**Pull request merged**